baseplate/context/cassandra.py

```diff
@@ -90,7 +96,50 @@ def make_object_for_context(self, name, server_span):
         return cqlmapper.connection.Connection(session_adapter)
 
 
-def _on_execute_complete(_, span):
+class WaitForCallbackResponseFuture(object):
+    """Wrap the ResponseFuture to ensure callbacks have completed.
+
+    This fixes a race condition where the server span can complete before
+    the callback has closed out the child span.
+
+    """
+
+    def __init__(self, span, future, callback_fn, callback_args, errback_fn, errback_args):
+        self.callback_event = Event()
+
+        def register_completion(fn):
```

Comment:
what do you think of passing the event object into `_on_execute_complete` / `_on_execute_failed` and letting them just directly call it explicitly rather than doing this wrapper stuff?

Reply:
the wrapper is kind of nice because it does the `try/finally`, but I'll try moving those to the callback methods and see how it looks